ui/packages/catalog-ui-search/src/main/webapp/react-component/container/theme-settings/theme-settings.tsx

```diff
@@ -0,0 +1,353 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global require*/
+import * as React from 'react'
+import styled from '../../../react-component/styles/styled-components'
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+const PropertyView = require('../../../component/property/property.view.js')
+const user = require('../../../component/singletons/user-instance.js')
+const Property = require('../../../component/property/property.js')
+const ThemeUtils = require('../../../js/ThemeUtils.js')
+
+const getFontSize = () => {
+  return getPreferences().get('fontSize')
+}
+
+const getPreferences = () => {
+  return user.get('user').get('preferences')
+}
+
+const getAnimationMode = () => {
+  return getPreferences().get('animation')
+}
+
+const getSpacingMode = () => {
+  return getPreferences()
+    .get('theme')
+    .getSpacingMode()
+}
+
+const getTheme = () => {
+  return getPreferences()
+    .get('theme')
+    .getColorMode()
+}
+
+const getCustomColorNames = () => {
+  return getPreferences()
+    .get('theme')
+    .getCustomColorNames()
+}
+
+const getCustomColors = () => {
+  return getCustomColorNames().map((colorVariable: any) => {
+    return new Property({
+      isEditing: true,
+      label: colorVariable,
+      value: [
+        getPreferences()
+          .get('theme')
+          .get(colorVariable),
+      ],
+      type: 'COLOR',
+    })
+  })
+}
+
+const Root = styled.div`
+  position: relative;
+  display: block;
+  text-align: center;
+`
+const ThemeCustom = styled.div`
+  display: block;
+`
+
+const saveFontChanges = (fontSize: any) => {
+  let preferences = getPreferences()
+  let newFontSize = fontSize
+  preferences.set('fontSize', ThemeUtils.getFontSize(newFontSize))
+}
+
+const saveSpacingChanges = (spacingMode: any) => {
```

Comment:
❓ Could the parameter in this function and in the functions below be more specifically typed? Or made into a type